### ISSUE TITLE
is it possible to open and read PDF report using pywinauto?

### ISSUE THREAD

**Issue by smitagodbole** [pywinauto/pywinauto#529]
Hi,

I would like to open a  report generated in PDF format and then read the data using pywinauto.  Do I need to import any library for this?  

Thanks
Smita

**Comment by vasily-v-ryabov** [comment]
Technically it's possible to use pywinauto to open PDF with Adobe Reader etc etc.
But I would recommend reading the PDF file directly using appropriate library: https://github.com/pmaupin/pdfrw

**Comment by smitagodbole** [comment]
thanks vasily !!.  I will read and try.
while working I found that I have some custom controls, table controls and pane control.  Is there any way that I can work with all these controls?  

Smita

**Comment by vasily-v-ryabov** [comment]
It depends how these controls are implemented. Standard table controls are supported in UIA backend. But usually they are visible as Pane. If such a control has `NativeWindowHandle` property in `Inspect.exe`, it's possible to monitor window messages handled by this control in Spy++. These window messages can be used to implement custom wrapper (generally it would look like standard wrappers for "win32" backend).

**Comment by vasily-v-ryabov** [comment]
In other words it's kind of reverse engineering if source code of these controls is not available. In the worst case authors of the custom controls could implement such window message handlers on app side to improve its test-ability. But this is mostly communication or management problem if anyone considers it as a problem of course.

**Comment by smitagodbole** [comment]
Hi Vasily,Thanks for reply.  Is this standard table control supports rows and columns?  I have few dialogs with table controls but haven't got success  with these rows and columns and test it.I am really confused with pane controller which is a custom control and there are many objects/elements inside this control which are not getting recognized.  Is there any example of this which I can see to understand more of this control.
Sorry, I am not a hard core developer.  I am a QA person who is trying to work on this tool.  Please bare with me if I am asking some dumb questions.
Smita

    On Tuesday, August 7, 2018, 11:45:32 AM EDT, Vasily Ryabov <notifications@github.com> wrote:  
 
 
In other words it's kind of reverse engineering if source code of these controls is not available. In the worst case authors of the custom controls could implement such window message handlers on app side to improve its test-ability. But this is mostly communication or management problem if anyone considers it as a problem of course.

**Comment by vasily-v-ryabov** [comment]
If all these tables are shown as Panes, this is pain. :)
What is a `ClassName` property values for these controls? Maybe some children class names are also important.
